src/_pytest/recwarn.py

```diff
@@ -268,16 +270,17 @@ def __exit__(
 
         __tracebackhide__ = True
 
+        def found_str():
+            return pformat([record.message for record in self], indent=2)
```

Comment:
Not sure if pformat makes sense for a flat list of strings. Wouldn't something like `"\n".join(f"- {record.message}" for record in self)` be more readable here?

Reply:
pformat will auto choose a compact/non-compact display based on the list given and the data lenght

additionally it ensures we see multi-line messages as what they are